Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
-							return;
-						}
+					// (4/4) IfcRectangularTrimmedSurface SUBTYPE of IfcBoundedSurface.
+					else if (surface.isOfType<typename IfcEntityTypesT::IfcRectangularTrimmedSurface>()) {
 
-						/*! \internal testing if the Cmake options to include internal comments works.
+						return convertIfcRectangularTrimmedSurface(
+							surface.as<typename IfcEntityTypesT::IfcRectangularTrimmedSurface>(),
+							pos,
+							polylineData);
+					}
+				}
+
+				//--------------------------------------------------------------------------------------------
+				// IfcElementarySurface
+				//--------------------------------------------------------------------------------------------
+
+						/*! \brief  Converts \c IfcElementarySurface by calling the conversion function according to subtype.
+						\param		surface	\c IfcElementarySurface entity to be interpreted.
+						\param		pos
+						\param		polylineData
+						\note		The \c IfcElementarySurface subtypes are \c IfcCylindricalSurface, \c IfcPlane, \c IfcSphericalPlane and \c IfcToroidalSurface.
 						*/
 
-						// (1/4) IfcBSplineSurface SUBTYPE of IfcBoundedSurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcBSplineSurface> bspline_surface =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcBSplineSurface>(bounded_surface);
-
-						if(bspline_surface) {
-							// Get attributes 1-4.
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcInteger> u_degree =
-								std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcInteger>(bspline_surface);*/
-							int u_degree = bspline_surface->UDegree;
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcInteger> v_degree =
-								std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcInteger>(bspline_surface);*/
-							int v_degree = bspline_surface->VDegree;
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcCartesianPoint> control_point_list =
-								std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcCartesianPoint>(bspline_surface);*/		// TO DO: next level (IfcCoordinates: IfcLengthMeasure)
-							auto vec_control_point_list_list = bspline_surface->ControlPointsList;
-							for(auto& it_control_point_list : vec_control_point_list_list) {
-								std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcCartesianPoint>> vec_control_point_list;
-								std::shared_ptr<ItemData> input_data_cpl_set(new ItemData);
-
-								vec_control_point_list.resize(it_control_point_list.size());
-								std::transform(it_control_point_list.begin(),
-									it_control_point_list.end(),
-									vec_control_point_list.begin(),
-									[](EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint> it) { return it.lock(); });
+				void convertIfcElementarySurface(const EXPRESSReference<typename IfcEntityTypesT::IfcBoundedSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {

```

Comment:
Please also change the out parameter here.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1